### PR TITLE
remove CAdvisorPort from KubeletConfiguration struct

### PR DESCRIPTION
**Pull request by mtaufen**
See: #56523, cAdvisor is becoming an implementation detail of
Kubernetes, and we should not canonize its knobs on the
KubeletConfiguration.

```release-note
NONE
```

### CODE REVIEW

File: cmd/kubelet/app/options/options.go [hunk 1]
```diff
@@ -232,6 +236,7 @@ func NewKubeletFlags() *KubeletFlags {
 		HostNetworkSources:  []string{kubetypes.AllSource},
 		HostPIDSources:      []string{kubetypes.AllSource},
 		HostIPCSources:      []string{kubetypes.AllSource},
+		CAdvisorPort:        4194,
```
Comment (luxas) [inline cmd/kubelet/app/options/options.go]: Can we add a TODO to change this to 0 in order to disable cAdvisor listening on the public port in the future?

File: cmd/kubelet/app/options/options.go [hunk 2]
```diff
@@ -115,6 +116,9 @@ type KubeletFlags struct {
 	// is true and upon the initial registration of the node.
 	RegisterWithTaints []core.Taint
 
+	// cAdvisorPort is the port of the localhost cAdvisor endpoint (set to 0 to disable)
+	CAdvisorPort int32
```
Comment (luxas) [inline cmd/kubelet/app/options/options.go]: Pros/cons regarding using a pointer or not here in the internal code?
Are we using the int32 to be consistent with most other fields or would uint16 make sense for this field internally for now?

**Review by luxas — APPROVED** [review]
LGTM, thanks @mtaufen!

File: cmd/kubelet/app/options/options.go (hunk 1 above)
Comment (mtaufen) [reply cmd/kubelet/app/options/options.go]: added

**Comment by mtaufen** [comment]
re-adding label after rebase

File: cmd/kubelet/app/options/options.go (hunk 2 above)
Comment (mtaufen) [reply cmd/kubelet/app/options/options.go]: Not sure of the details of how Uint16Var flags get parsed. Do you get an error if you overflow it? I'd imagine they'd just use strconv.ParseUint under the hood, which should just give you a "value out of range" error? If so we could skip this in our validation.

**Pull request merged**